### PR TITLE
fix(amplify_analytics): fix analyticsProperties

### PR DESCRIPTION
**Pull request by fjnoyp**
*Issue #, if available:*

*Description of changes:*

Fix sending of AnalyticsProperties in iOS 

Before iOS was converting Dart maps representing AnalyticsProperties to blank maps every time. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_analytics_pinpoint/android/src/main/kotlin/com/amazonaws/amplify/amplify_analytics_pinpoint/AmplifyAnalyticsBuilder.kt [hunk 1]
```diff
@@ -98,10 +98,13 @@ class AmplifyAnalyticsBuilder {
...
                         userProfileBuilder.location(createUserLocation(locationMap))
                     }
-                    "properties" -> {
+                    "propertiesMap" -> {
                         val propertiesMap = item.value as HashMap<String, Any>
                         userProfileBuilder.customProperties(createAnalyticsProperties(propertiesMap))
                     }
+                    "propertiesTypesMap" -> {
+                        // Ignore, we can infer types just using propertiesMap
+                    }
```
Comment (Amplifiyer) [inline packages/amplify_analytics_pinpoint/android/src/main/kotlin/com/amazonaws/amplify/amplify_analytics_pinpoint/AmplifyAnalyticsBuilder.kt:107]: probably a good idea to do it anyways? I've seen in DataStore `1` being translated to `1.0` after traveling through method channel serialization. If that didn't happen in your testing, then we can keep it as is.

File: packages/amplify_analytics_pinpoint/ios/Classes/AmplifyAnalyticsBuilder.swift [hunk 2]
```diff
@@ -18,8 +18,27 @@ import AmplifyPlugins
...
+                    analyticsProperties[key] = propertiesMap[key] as! String
+                case "INT":
+                    analyticsProperties[key] = propertiesMap[key] as! Int
+                case "DOUBLE":
+                    analyticsProperties[key] = propertiesMap[key] as! Double
+                case "BOOL":
+                    analyticsProperties[key] = propertiesMap[key] as! Bool
+                default:
+                    print("Unknown type for AnalyticsProperties")
+            }
```
Comment (Amplifiyer) [inline packages/amplify_analytics_pinpoint/ios/Classes/AmplifyAnalyticsBuilder.swift:39]: If we can avoid force unwrap, we should do that

File: packages/amplify_analytics_pinpoint/android/src/main/kotlin/com/amazonaws/amplify/amplify_analytics_pinpoint/AmplifyAnalyticsBuilder.kt (hunk 1 above)
Comment (fjnoyp) [reply packages/amplify_analytics_pinpoint/android/src/main/kotlin/com/amazonaws/amplify/amplify_analytics_pinpoint/AmplifyAnalyticsBuilder.kt:107]: hmm what are you referring to?  We ignore this case as it's already handled for the propertiesMap case.  

I've verified that for Android a double property sent as 10.0 and an int property sent as 10 are saved as their proper types.

Comment (fjnoyp) [reply packages/amplify_analytics_pinpoint/android/src/main/kotlin/com/amazonaws/amplify/amplify_analytics_pinpoint/AmplifyAnalyticsBuilder.kt:107]: iOS was a bit trickier hence the extra data structure propertiesTypesMap being sent from Dart

Comment (Amplifiyer) [reply packages/amplify_analytics_pinpoint/android/src/main/kotlin/com/amazonaws/amplify/amplify_analytics_pinpoint/AmplifyAnalyticsBuilder.kt:107]: I was referring to inferring the types in android. There were some issues in DataStore when `int` were being converted to `double` by GSON library. If that's not the case here, then we can ignore it.

File: packages/amplify_analytics_pinpoint/ios/Classes/AmplifyAnalyticsBuilder.swift (hunk 2 above)
Comment (fjnoyp) [reply packages/amplify_analytics_pinpoint/ios/Classes/AmplifyAnalyticsBuilder.swift:39]: propertiesTypesMap needs to be force unwrapped as it's used for the foreach.  But I can use ? for String/Int/Double/Bool casts.

**Review by Amplifiyer — APPROVED** [review]

**Pull request merged**